corehq/ex-submodules/dimagi/utils/retry.py

```diff
@@ -2,25 +2,32 @@
 from time import sleep
 
 
-def retry_on(*errors, should_retry=lambda err: True):
+def retry_on(
+    *errors,
+    should_retry=lambda err: True,
+    delays=(0.1, 1, 2, 4, 8, None),
+):
     """Make a decorator to retry function on any of the given errors
 
     Retry up to 5 times with exponential backoff. Raise the last
     received error if all calls fail.
 
     :param *errors: One or more errors (exception classes) to catch and retry.
     :param should_retry: Optional function taking one argument, an
-    exception instance, that returns true to signal a retry and false
-    to signal that the error should be re-raised.
+        exception instance, that returns true to signal a retry and false
+        to signal that the error should be re-raised.
+    :param delays: An iterable of delays given in seconds. It must
+        terminate in ``None`` to raise an error in ``errors``, otherwise
+        it will stop retrying silently.
```

Comment:
I'm not sure we want expose this "feature" because it is easy to miss and would have a bad outcome (error would pass silently) if overlooked. What do you think of automatically adding `None`? Something like
```py
from itertools import chain

for delay in chain(delays, [None]):
```